playwright/src/test/java/com/microsoft/playwright/TestLocatorAssertions.java

```diff
@@ -165,6 +165,66 @@ void hasTextWTextArrayFailOnNotEmpty() {
     }
   }
 
+  @Test
+  void defaultTimeoutWithValueZeroPass() {
+    LocatorAssertions.TimeoutOptions.setDefaultTimeout(0);
+    page.setContent("<div id=div>Text 1</div>");
+    Locator locator = page.locator("#div");
+    assertThat(locator).hasText("Text 1");
+  }
+
+  @Test
+  void defaultTimeoutWithValueZeroFail() {
+    LocatorAssertions.TimeoutOptions.setDefaultTimeout(0);
+    page.setContent("<div id=div></div>");
+    Locator locator = page.locator("p");
+    page.evaluate("setTimeout(() => {\n" +
+      "  div.innerHTML = \"<p>Text 1</p><p>Text 2</p>\";\n" +
+      "}, 100);");
+    try {
+      // Should normalize whitespace.
+      assertThat(locator).hasText(new String[] {"Text  1", "Text   2"});
+      fail("did not throw");
+    } catch (AssertionFailedError e) {
+      assertEquals("[Text  1, Text   2]", e.getExpected().getStringRepresentation());
+      assertEquals("null", e.getActual().getStringRepresentation());
```

Comment:
@yury-s @mxschmitt I could use some help here.  `e.getExpected()`returns null when the default timeout is set to 0.  Any value great than 0 will work as expected but i'm not sure why that is.

Reply:
`getExpected()` and `getActual()` both return null:

<img width="648" alt="image" src="https://user-images.githubusercontent.com/1053584/158726223-3193d0ca-0d41-41be-a0d1-48dd36d8a6a2.png">